detectron2/evaluation/wandb.py

```diff
@@ -0,0 +1,320 @@
+                        "box_caption": "%s %.3f" % (caption, pred["scores"][i]),
+                        "scores": {"class_score": pred["scores"][i]},
+                        "domain": "pixel",
+                    }
+                )
+
+                avg_conf_per_class[pred_class] = avg_conf_per_class[pred_class] + pred["scores"][i]
+                if pred_class in counts:
+                    counts[pred_class] = counts[pred_class] + 1
+                else:
+                    counts[pred_class] = 1
+
+            for pred_class in counts.keys():
+                avg_conf_per_class[pred_class] = avg_conf_per_class[pred_class] / counts[pred_class]
+
+            boxes = {
+                "predictions": {
+                    "box_data": boxes_data,
+                    "class_labels": self.thing_index_to_class,
+                }
+            }
+
+        masks = {}
+        # pixles_per_class = [0 for _ in self.stuff_class_names]
+        # Process semantic segmentation predictions
+        if pred.get("sem_mask") is not None:
+            masks["semantic_mask"] = {
+                "mask_data": pred["sem_mask"],
+                "class_labels": self.stuff_index_to_class,
+            }
+            classes = self._table_stuff_classes
+
+        # TODO: Support panoptic , instance segmentation maks & keypoint visualizations.
+        '''
+        # Process instance segmentation detections
+        if pred.get("pred_masks") is not None:
+            class_count = {}
+            num_pred = min(15, len(pred["pred_masks"]))  # Hardcoded to max 15 masks for better UI
+            for i in range(num_pred):
+                pred_class = int(pred["classes"][i])
+                if pred_class in class_count:
+                    class_count[pred_class] = class_count[pred_class] + 1
+                else:
+                    class_count[pred_class] = 0
+
+                # title format - class_count. E.g - person_0, person_1 ..
+                mask_title = (
+                    f"class {pred_class}"
+                    if not self.thing_class_names[loader_i]
+                    else self.thing_class_names[loader_i][pred_class]
+                )
+                mask_title = f"{mask_title}_{class_count[pred_class]}"
+
+                """
+                masks[mask_title] = {
+                    "mask_data": pred['pred_masks'][i]*(pred_class+1),
+                    "class_labels": {pred_class+1: mask_title}
+                }
+                """
+
+        if pred.get("panoptic_mask") is not None:
+            masks["panoptic_mask"] = {
+                "mask_data": pred["panoptic_mask"],
+                "class_labels": self.stuff_index_to_class[loader_i],
+            }
+        '''
+
+        table_row = [
+            file_name,
+            self.wandb.Image(file_name, boxes=boxes, masks=masks, classes=classes),
+        ]
+        if pred.get("boxes") is not None:
+            table_row.extend(avg_conf_per_class)
```

Comment:
@ppwwyyxx  Thanks for the insight. I guess It'll be a UI problem to display conf. each prediction of each class as unique value so I used this less useful metric. At the most, it can provide info if a model is doing bad in a particular class. 
It looks like [this](https://wandb.ai/cayush/pr_review/reports/Shared-panel-22-01-05-14-01-11--VmlldzoxNDA1ODAx) in a workspace. 
Do you think we can include some other metrics to make this table more useful for practitioners/researchers?